src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -644,19 +644,42 @@ private static void AddPackageDownloads(PackageSpec spec, IEnumerable<IMSBuildIt
             foreach (var item in GetItemByType(items, "DownloadDependency"))
             {
                 var id = item.GetProperty("Id");
-                var versionRange = GetVersionRange(item);
+                var versionString = item.GetProperty("VersionRange");
 
-                if (!(versionRange.HasLowerAndUpperBounds && versionRange.MinVersion.Equals(versionRange.MaxVersion)))
+                // If there are multiple package download elements in the project for the same package id, only the first is used.
+                // But, the project might multi-target, and the package download is for one target, so we must check each tfm.
+                var frameworks = GetFrameworks(item).ToList();
+                for (int i = 0; i < frameworks.Count; i++)
```

Comment:
I'm not sure what you are trying to accomplish here. 
It feels unnecessarily convoluted. 

You should only change how the VersionRange is parsed.

Reply:
You should do the deduping in the reading task for simplicity.